src/main/java/com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser.java

```diff
@@ -443,6 +462,50 @@ private static String getNodeClassNameWithoutContext(ParseTree node) {
     }
 
     /**
+     * Method to generate an error message about missing HTML tag if the ANTLR error is indeed
+     * because of the missing tag.
+     * @param exception {@code NoViableAltException} object catched while parsing javadoc
+     * @return returns appropriate {@code ParseErrorMessage} if a HTML close tag is missed;
+     *     null otherwise
+     */
+    private ParseErrorMessage generateMissedHtmlTagErrorMessage(RecognitionException exception) {
+        ParseErrorMessage parseErrorMessage = null;
+        final Interval sourceInterval = exception.getCtx().getSourceInterval();
+        final List<Token> tokenList;
+        tokenList = ((BufferedTokenStream) exception.getInputStream())
+                .getTokens(sourceInterval.a, sourceInterval.b);
+        final ArrayDeque<Token> stack = new ArrayDeque<>();
+        Token token;
+        Token htmlTagNameStart;
```

Comment:
Please move declaration of these two variables to the place of their initialization.